core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -399,15 +407,51 @@ object CachedDataFrame
       row.writeToStream(bufferOutput, buffer)
       count += 1
     }
+
     flushBufferOutput(bufferOutput, bufferOutput.position(), output, codec)
-    if (count > 0) {
-      if (output.position() == output.getBuffer.length) {
-        new PartitionResult(output.getBuffer, count)
+    var memSize = 0L
+    var enoughMemory = true
+    try {
+      if (count > 0) {
+        val finalBuffer = output.getBufferRetain
+        finalBuffer.flip
+        memSize = finalBuffer.limit()
+        // Ask UMM before getting the array to heap.
+        // Taking unroll memory as this memory is cleaned up on task completion.
+        // On connector mode also this should account to the overall memory usage.
+        // We will ensure that sufficient memory is available by reserving
+        // two times as Kryo serialization will expand its buffer accordingly.
+
+        if (!MemoryManagerCallback.memoryManager.
+          acquireStorageMemoryForObject(
```

Comment:
Problem was, I wanted a memory allocation which should not evict data anymore. Right now our UMM acquireExecutionMemory() does not have that flag. I will have to change that before I can do this.